Tests/TuistKitTests/Cache/CacheControllerTests.swift

```diff
@@ -83,19 +91,86 @@ final class CacheControllerTests: TuistUnitTestCase {
             XCTAssertEqual(loadPath, path)
             return (xcworkspacePath, graph)
         }
+        generator.generateStub = { (loadPath, _) -> AbsolutePath in
+            XCTAssertEqual(loadPath, path)
+            return xcworkspacePath
+        }
+        graphContentHasher.stubbedContentHashesResult = nodeWithHashes
+        artifactBuilder.stubbedCacheOutputType = .xcframework
+
+        // When
+        try subject.cache(path: path, configuration: "Debug", targetsToFilter: [])
+
+        // Then
+        XCTAssertPrinterOutputContains("""
+        Hashing cacheable targets
+        Building cacheable targets
+        Building cacheable targets: \(aTarget.name), 1 out of 3
+        Building cacheable targets: \(bTarget.name), 2 out of 3
+        Building cacheable targets: \(cTarget.name), 3 out of 3
+        All cacheable targets have been cached successfully as xcframeworks
+        """)
+        XCTAssertEqual(cacheGraphLinter.invokedLintCount, 1)
+        XCTAssertEqual(artifactBuilder.invokedBuildWorkspacePathParametersList[0].target, aTarget)
+        XCTAssertEqual(artifactBuilder.invokedBuildWorkspacePathParametersList[1].target, bTarget)
+        XCTAssertEqual(artifactBuilder.invokedBuildWorkspacePathParametersList[2].target, cTarget)
+    }
+
+    func test_filtered_cache_builds_and_caches_the_frameworks() throws {
+        // Given
+        let path = try temporaryPath()
+        let xcworkspacePath = path.appending(component: "Project.xcworkspace")
```

Comment:
nit: Seems like some of this logic could be extracted in funcs

Reply:
Can you share more details? To which logic are you referring too? (I tried to extract the sut but different models are needed graph, path, and the targets)

Reply:
I meant a func for creating the a/b/c target nodes, but this isn't anything substantial, go ahead and merge when you're ready!